src/extension.ts

```diff
@@ -96,11 +122,14 @@ function fetchAndLaunchMetals(context: ExtensionContext, javaHome: string) {
   if (dottyArtifact && fs.existsSync(dottyArtifact)) {
     outputChannel.appendLine(
       `Metals will not start since Dotty is enabled for this workspace. ` +
-        `To enable Metals, remove the file ${dottyArtifact} and run 'Reload window'`
+      `To enable Metals, remove the file ${dottyArtifact} and run 'Reload window'`
     );
     return;
   }
 
+  if (!javaHome || javaHome.length == 0 || !fs.existsSync(javaHome))
+    throw new Error("Invalid Java Home path");
```

Comment:
Should we report an error message to the user instead of throwing an exception?

Reply:
It gets reported to user when it's caught by catch at line 82